crates/symbolicli/src/main.rs

```diff
@@ -0,0 +1,655 @@
+
+            table.add_row(row![r->"  File:", format!("{file}:{line}")]);
+        }
+
+        table.add_empty_row();
+    }
+    table.printstd();
+}
+
+#[derive(Debug)]
+enum Payload {
+    Event(event::Event),
+    Minidump(TempPath),
+}
+
+impl Payload {
+    async fn process(
+        self,
+        symbolication: &SymbolicationActor,
+        scope: Scope,
+        sources: Arc<[SourceConfig]>,
+    ) -> anyhow::Result<CompletedSymbolicationResponse> {
+        match self {
+            Payload::Event(event) => {
+                let symbolication_request =
+                    event::create_symbolication_request(scope, sources, event)
+                        .context("Event cannot be symbolicated")?;
+
+                tracing::info!("symbolicating event");
+                symbolication.symbolicate(symbolication_request).await
+            }
+            Payload::Minidump(minidump_path) => {
+                tracing::info!("symbolicating minidump");
+                symbolication
+                    .process_minidump(scope, minidump_path, sources)
+                    .await
+            }
+        }
+    }
+
+    fn parse<P: AsRef<Path> + fmt::Debug>(path: &P) -> anyhow::Result<Option<Self>> {
+        match std::fs::File::open(path) {
+            Ok(mut file) => {
+                let mut magic = [0; 4];
+                file.read_exact(&mut magic)?;
+                file.seek(SeekFrom::Start(0))?;
+
+                if &magic == b"MDMP" || &magic == b"PMDM" {
+                    let mut temp_file = NamedTempFile::new().unwrap();
+                    std::io::copy(&mut file, &mut temp_file)
+                        .context("Failed to write minidump to disk")?;
+
+                    Ok(Some(Payload::Minidump(temp_file.into_temp_path())))
+                } else {
+                    let event =
+                        serde_json::from_reader(file).context("failed to parse event json file")?;
+                    Ok(Some(Payload::Event(event)))
+                }
+            }
+            Err(e) if e.kind() == std::io::ErrorKind::NotFound => Ok(None),
+            Err(e) => Err(e).context(format!("Could not open event file at {:?}", path)),
+        }
+    }
+
+    async fn get_remote(
+        client: &reqwest::Client,
+        base_url: &Url,
+        org: &str,
+        project: &str,
+        event_id: &str,
+    ) -> anyhow::Result<Self> {
+        tracing::info!("trying to resolve event remotely");
+        let event = remote::download_event(client, base_url, org, project, event_id).await?;
+        tracing::info!("event json file downloaded");
+
+        match remote::get_attached_minidump(client, base_url, org, project, event_id).await? {
+            Some(minidump_url) => {
+                tracing::info!("minidump attachment found");
+                let minidump_path = remote::download_minidump(client, minidump_url).await?;
+                tracing::info!(path = ?minidump_path, "minidump file downloaded");
```

Comment:
What's `?` when used as a prefix? 👀

Reply:
That's specific to `tracing`. It just means to use `Debug` for formatting.